build.gradle

```diff
@@ -3,178 +3,130 @@
+      }
+    }
 
-task testJar(type: Jar, dependsOn: testClasses) {
-    classifier = 'tests'
-    from sourceSets.test.output
-}
+    checkstyle {
+      configDir = rootProject.file('config/checkstyle')
+      toolVersion = '8.4'
+    }
 
-build.dependsOn 'testJar'
-jacocoTestReport.dependsOn test
-project.tasks["sonarqube"].dependsOn jacocoTestReport
+    test.dependsOn('checkstyleMain', 'checkstyleTest')
 
-configurations {
-    testArtifacts.extendsFrom testRuntime
-}
+    task sourcesJar(type: Jar, dependsOn: classes) {
+      classifier = 'sources'
+      from sourceSets.main.allSource
+    }
 
-artifacts {
-    testArtifacts testJar
-    archives sourcesJar
-    archives javadocJar
-}
+    task javadocJar(type: Jar, dependsOn: javadoc) {
+      classifier = 'javadoc'
+      from javadoc.destinationDir
+    }
 
-//maven artifact(s) generation
-
-publishing {
-    publications {
-        standardjava(MavenPublication) {
-            from components.java
-            artifact sourcesJar
-            artifact javadocJar
-            artifact testJar
-            pom.withXml {
-                asNode().children().last() + {
-                    resolveStrategy = Closure.DELEGATE_FIRST
-                    description 'li-apache-kafka-clients'
-                    url 'https://github.com/linkedin/li-apache-kafka-clients'
-                    scm {
-                        url 'https://github.com/linkedin/li-apache-kafka-clients'
-                        connection 'scm:git:git://github.com/linkedin/li-apache-kafka-clients.git'
-                        developerConnection 'scm:git:ssh:git@github.com:linkedin/li-apache-kafka-clients.git.git'
-                    }
-                    licenses {
-                        license {
-                            name 'BSD 2-CLAUSE LICENSE'
-                            url 'https://opensource.org/licenses/BSD-2-Clause'
-                            distribution 'repo'
-                        }
-                    }
-                }
-            }
-            pom.withXml {
-                asNode().children().get(4).children().remove(0) //clear dependencies
-                asNode().children().get(4).appendNode("dependency")
-                .appendNode("groupId", "org.apache.kafka").parent()
-                .appendNode("artifactId", "kafka-clients").parent()
-                .appendNode("version", "0.10.2.1").parent()
-                .appendNode("scope", "compile").parent()
-                .parent()
-                .appendNode("dependency")
-                .appendNode("groupId", "commons-io").parent()
-                .appendNode("artifactId", "commons-io").parent()
-                .appendNode("version", "2.5").parent()
-                .appendNode("scope", "test").parent()
-            }
-        }
+    artifacts {
+      archives sourcesJar
+      archives javadocJar
     }
+  }
 }
 
-//publishing (artifactory)
-
+artifactoryPublish.skip = true
```

Comment:
Does it mean that the automated publications are disabled. Is it intentional?

Reply:
i think this just disables publishing the parent project. im not sure as ethan gave me this part of the conf